src/main/java/software/amazon/encryption/s3/S3EncryptionClient.java

```diff
@@ -396,7 +423,13 @@ public CreateMultipartUploadResponse createMultipartUpload(CreateMultipartUpload
     @Override
     public UploadPartResponse uploadPart(UploadPartRequest request, RequestBody requestBody)
             throws AwsServiceException, SdkClientException {
-        return _multipartPipeline.uploadPart(request, requestBody);
+        try {
+            return _multipartPipeline.uploadPart(request, requestBody);
+        } catch (CompletionException e) {
+            throw new S3EncryptionClientException(e.getCause().getMessage(), e.getCause());
+        } catch (Exception e) {
+            throw new S3EncryptionClientException("Unable to delete objects.", e);
```

Comment:
Same here, the error message here is wrong for this method `uploadPart`. Instead, we should be throwing `e.getMessage()`.